src/main/java/org/alfresco/repo/jscript/RhinoScriptProcessor.java

```diff
@@ -643,27 +643,27 @@ public void afterPropertiesSet() throws Exception
      *                  is to be reused.
      * @return the scope object
      */
-    protected Scriptable initScope(Context cx, boolean secure, boolean sealed)
-    {
-        Scriptable scope;
-        if (secure)
-        {
-            // Initialise the non-secure scope
-            // allow access to all libraries and objects, including the importer
-            // @see http://www.mozilla.org/rhino/ScriptingJava.html
-            scope = new ImporterTopLevel(cx, sealed);
-        }
-        else
-        {
-            // Initialise the secure scope
-            scope = cx.initStandardObjects(null, sealed);
-            // remove security issue related objects - this ensures the script may not access
-            // unsecure java.* libraries or import any other classes for direct access - only
-            // the configured root host objects will be available to the script writer
-            scope.delete("Packages");
-            scope.delete("getClass");
-            scope.delete("java");
-        }
-        return scope;
+    protected Scriptable initScope(Context cx, boolean secure, boolean sealed)
+    {
+        Scriptable scope;
+        if (secure)
+        {
+            // Initialise the non-secure scope
+            // allow access to all libraries and objects, including the importer
+            // @see http://www.mozilla.org/rhino/ScriptingJava.html
+            scope = new ImporterTopLevel(cx, sealed);
+        }
```

Comment:
What is the meaning of the "secure" parameter here? Shouldn't it be the other way round?

Reply:
I didn't change nothing on that part. The meaning of secure in this case is that we consider code to be secure for exemple using javascript as string or javascript files from classPath. I didn't change this part. If you take a look you will se that I change (insecure) using Rhino  initSafeStandardObjects this is used for javascript inside nodeRef. Please see: https://mozilla.github.io/rhino/javadoc/org/mozilla/javascript/Context.html#initSafeStandardObjects-org.mozilla.javascript.ScriptableObject.

Reply:
The main idea is that Rhino introduce from 1.2.7 the concept of sandbox as you mentioned in the jira and I used. I didn't add parameters because I repaired the delete("java") with the call of initSafe.....